src/main/java/org/elasticsearch/index/mapper/core/IntegerFieldMapper.java

```diff
@@ -197,9 +199,7 @@ public Query fuzzyQuery(String value, Fuzziness fuzziness, int prefixLength, int
 
     @Override
     public Query termQuery(Object value, @Nullable QueryParseContext context) {
-        int iValue = parseValue(value);
-        return NumericRangeQuery.newIntRange(names.indexName(), precisionStep,
-                iValue, iValue, true, true);
+        return new TermQuery(new Term(names.indexName(), indexedValueForSearch(value)));
```

Comment:
Could we move this up to `NumberFieldMapper` (replacing the range query there now) to get the benefit with the other types @rmuir mentions?